File: backend/src/main/kotlin/org/icpclive/events/codeforces/api/data/CFParty.java
package org.icpclive.events.codeforces.api.data;

import java.util.List;

/**
 * @author egor@egork.net
 */
public class CFParty {
    public Integer contestId;
    public List<CFMember> members;
    public CFPartyParticipantType participantType;
    public Integer teamId;
    public String teamName;
    public boolean ghost;
    public Integer room;
    public Long startTimeSeconds;

    public enum CFPartyParticipantType {
        CONTESTANT, PRACTICE, VIRTUAL, MANAGER, OUT_OF_COMPETITION
    }
}

File: backend/src/main/kotlin/org/icpclive/cds/codeforces/api/data/CFParty.kt
@file:Suppress("UNUSED")
package org.icpclive.cds.codeforces.api.data

import kotlinx.serialization.*

enum class CFPartyParticipantType {
    CONTESTANT, PRACTICE, VIRTUAL, MANAGER, OUT_OF_COMPETITION
}

@Serializable
data class CFParty(
    val contestId: Int? = null,
    val members: List<CFMember>,
    val participantType: CFPartyParticipantType,
    val teamId: Int? = null,
    val teamName: String? = null,
    val ghost: Boolean,
    val room: Int? = null,
    val startTimeSeconds: Long? = null,
)